api/mutations/thread/publishThread.js

```diff
@@ -162,14 +226,67 @@ export default async (
     });
   }
 
-  // $FlowIssue
-  const dbThread = await publishThread(threadObject, currentUser.id);
+  // $FlowFixMe
+  const dbThread: DBThread = await publishThread(threadObject, currentUser.id);
+
+  // we check for toxicity here only to determine whether or not to send
+  // email notifications - the thread will be published regardless, but we can
+  // prevent some abuse and spam if we ensure people dont get email notifications
+  // with titles like "fuck you"
+  const threadIsToxic = async () => {
+    const body = thread.content.body
+      ? threadBodyToPlainText(thread.content.body)
+      : '';
+    const title = thread.content.title;
+    const text = `${title} ${body}`;
+
+    const scores = await Promise.all([
+      getSpectrumScore(text, dbThread.id, dbThread.creatorId),
+      getPerspectiveScore(text),
+    ]).catch(err =>
+      console.error(
```

Comment:
We should make sure to properly handle the error case here b/c this actually happens fairly often if you look at the athena logs (~once per hour)

Reply:
It's just random Perspective failures, no idea why - catching the error here should still keep the overall mutation safe right?